### ISSUE TITLE
Error when "session context" is set to "database table"

### ISSUE THREAD

**Issue by FrDH** [solspace/craft3-freeform#599]
**Describe the bug or issue you're experiencing**
When I set the "Freeform Session Context" option (in the "Form Behavior" tab) to "Using a database table", submitting the form gives an error for `$bag->getFormId(),` (see below)

**Steps to reproduce**

1. Set the Session Context option to use a database table.
2. Create a form and render it on a page
3. Submit the form

**Expected behavior**
I'd expect it not to throw an error :)

**Craft & Plugin Info (please complete the following information):**

- Craft Version: 3.7.63.1
- Freeform Version: 3.13.22.1
- Freeform Edition: PRO
- Fresh Install or Upgrade: Upgrade

**Additional context**
```
Call to a member function getFormId() on null
in /Users/path/to/website.nl/vendor/solspace/craft-freeform/packages/plugin/src/Bundles/Form/Context/Session/StorageTypes/DatabaseStorage.phpat line 75
 
  foreach ($this->context as $key => $bag) {
      $isNew = !\in_array($key, $existingKeys, true);
 
      $payload = [
          'contextKey' => $key,
          'sessionId' => \Craft::$app->session->getId(),
          'formId' => $bag->getFormId(),
          'propertyBag' => json_encode($bag->getProperties()),
          'attributeBag' => json_encode($bag->getAttributes()),
          'dateUpdated' => $bag->getLastUpdate(),
      ];
 
      if ($isNew) {
          \Craft::$app->db
              ->createCommand()
              ->insert(SessionContextRecord::TABLE, $payload)
```

**Comment by kjmartens** [comment]
Thanks for reporting this @FrDH. I can duplicate the issue on my end and we'll work on a fix for it shortly. 🙂

**Comment by kjmartens** [comment]
We've fixed this in v3 and v4 (was only able to reproduce in v4 though). If you wish to try this out right away on v3, you can try updating Freeform to `dev-master as 3.13.24.1` for the version. 🙂

**Comment by FrDH** [comment]
Hi,

Sorry for the late reply, must have missed the notification-mail :/

I'm not all that familiar with this kind of versioning, so I might be doing something wrong, but I get the following error when trying to install freeform using `"solspace/craft-freeform": "dev-master as 3.13.24.1"`:
```  
Problem 1
    - Root composer.json requires solspace/craft-freeform dev-master as 3.13.24.1, found solspace/craft-freeform[dev-fix/SFT-294, 2.0.0-beta.1, ..., v2.x-dev, 3.0.0-beta.1, ..., v3.x-dev, 4.0.0-beta.1, ..., v4.x-dev, v5.x-dev] but it does not match the constraint.
```

**Comment by kjmartens** [comment]
This has already been released normally so you can just update regularly now. 🙂
